### PR TITLE
Ensure all source nodes have resolved their paths.

### LINKED ISSUE #8: Should transform paths into absolute paths.
Given any arbitrary input path in the working directory, we should expand relative paths to their absolute path.

Without this, any plugins must **also** use the same `process.cwd()` when reading from their own `this.inputPaths`.

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes #8

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -32,12 +34,16 @@ class Directory {
...
+  constructor(_directoryPath: string, watched?: boolean | string, options?: DirectoryOptions) {
+    if (typeof _directoryPath !== 'string') {
+      throw new Error('Expected a path (string), got ' + _directoryPath);
     }
 
-    this._directoryPath = directoryPath;
+    // toNamespacedPath was added in Node 9, this guard can be removed after
+    // dropping support for Node < 10
+    let directoryPath = path.toNamespacedPath ? path.toNamespacedPath(_directoryPath) : _directoryPath;
+    this._directoryPath = path.resolve(directoryPath);
```
Comment (rwjblue) [inline index.ts:42]: Note: even after we can drop the guard for `toNamespacedPath` we will still need to do this `path.resolve`. Specifically, on POSIX systems `toNamespacedPath` always returns the input path unmodified, but on Windows `toNamespacePath` normalizes. 😭

Comment (thoov) [reply index.ts:42]: D23ZcurWkAA3zRG

File: test/unit.test.ts [hunk 2]
```diff
@@ -1,20 +1,62 @@
...
+        name: 'SomeName',
+        annotation: 'some annotation',
+      });
+
+      let pluginInterface = node.__broccoliGetInfo__();
+
+      expect(pluginInterface).to.have.property('sourceDirectory', resolvePath('some/path'));
+
+      if (os.platform() === 'win32' && semver.gte(process.version, '9.0.0')) {
+        expect(pluginInterface.sourceDirectory).to.not.equal(inputPath);
```
Comment (thoov) [inline test/unit.test.ts]: Could we add a comment or message for this case

Comment (rwjblue) [reply test/unit.test.ts]: I removed the toNamespacedPath stuff for now (windows CI failed with Broccoli 0.16). I'll be bringing it back in a separate PR.

**Comment by rwjblue** [comment]
This is probably ready for another look. I dropped the `toNamespacedPath` stuff for now, as it had issues with Broccoli 0.16 + Node 10 + Windows, I'll be working to bring it back subsequently (because I do think its more appropriate).

**Review by thoov — APPROVED** [review]

**Pull request merged**